### ISSUE TITLE
Add a Swagger schema

### ISSUE THREAD

**Issue by rhuss** [jolokia/jolokia#80]
It would be nice to have a [Swagger](http://developers.helloreverb.com/swagger/) schema for the Jolokia protocol, which could help in documenting it and trying it out.

You should look at the [examples](https://github.com/wordnik/swagger-core/tree/master/samples/no-server) since the official [documentation](https://github.com/wordnik/swagger-core/wiki/API-Declaration) seems to be a bit outdated.

More Links
- Product Paget: http://developers.helloreverb.com/swagger/
- Demo: http://petstore.swagger.wordnik.com/

**Comment by abhigupta-ny** [comment]
This would be a great feature to have. Besides being able to generate clients in all languages that swagger supports, it also provides a convenient web-gui to execute the rest calls. So you get an out of the box JMX web GUI for your application. 

On a slightly separate note, is there some simple light-weight HTML gui for jolokia you'd recommend that I can host along side it which gives me an html interface to jolokia. I see you've recommended hawtio somewhere - I haven't looked at it in detail yet but seems too heavyweight, whereas I'm looking for something very cheap probably a single/few html/js files that understand the jolokia interface. That's really what I'm after atm.

Thanks,
Abhi

**Comment by rhuss** [comment]
Unfortunately, there is none yet. There was one planned called Aji, but it never took off.

sorry.

**Comment by DebbautSara** [comment]
Any progress on this?

**Comment by rhuss** [comment]
@DebbautSara sorry, there are no plans yet to implement that but would be happy to support any Pull Request for this feature.

**Comment by grgrzybek** [comment]
Personally I think Jolokia's JSON/HTTP -> JMX [_protocol adaptor_](https://docs.oracle.com/en/java/javase/17/jmx/jmx-technology-architecture.html) is too low level to provide OpenAPI schema for it.
It's supposed to provide an adaptation layer, not a real API.
There are libraries that provide actual API on top of Jolokia, like https://github.com/artemiscloud/activemq-artemis-jolokia-api-server with their own OpenAPI definitions.

WDYT @rhuss @tadayosi ? I'd close this issue.

**Comment by tadayosi** [comment]
I haven't thought about it in depth, but I just wonder if it's possible to express Jolokia API with the OpenAPI schema considering Jolokia is not really a REST API.

**Comment by grgrzybek** [comment]
That's the point ;)

**Comment by rhuss** [comment]
Yeah, I wouldn't bother too much. Feel free to close this 🦖 issue :-)

**Comment by grgrzybek** [comment]
One more issue and we'll get to only two pages of issues (2x25)!
